pydantic/main.py

```diff
@@ -277,6 +277,7 @@ def is_untouched(v: Any) -> bool:
             '__slots__': slots | private_attributes.keys(),
             '__hash__': hash_func,
             '__class_vars__': class_vars,
+            '__match_args__': tuple(fields),
```

Comment:
I'm actually not sure we should add this, without it pattern matching still works, just not with positional arguments.

Since pydantic models don't support positional arguments, I think adding pseudo support for them in pattern matching could be very confusing.

@PrettyWood what do you think?

Either way, the rest of this PR would still be a great addition to pydantic.

Reply:
Ah I thought this was needed! If kwargs matching already works I reckon we shouldn't add this. Allowing positional arguments only here but not elsewhere is confusing

Reply:
Agreed.

That was my assumption, but I tried it and it seems to work fine without this.

Reply:
I agree. Even though it peforms a completely different function, the match-case syntax looks the same as instantiation, so it looks offputting given that BaseModel doesn't use args.